src/main/webapp/app/exam/manage/students/upload-images/students-upload-images-dialog.component.ts

```diff
@@ -37,8 +37,8 @@ export class StudentsUploadImagesDialogComponent implements OnDestroy {
     notFoundUsers?: NotFoundExamUserType;
     file: File;
 
-    courseId = input.required<number>();
-    exam = input.required<Exam>();
+    courseId: number;
+    exam: Exam;
```

Comment:
Why don't you annotate it with `@Input`? So it is clear that this value needs to be provided on usage